download_weather_incremental.py

```
@@ -0,0 +1,366 @@
+#!/usr/bin/env python3
+
+import argparse
+import asyncio
+import sys
+from datetime import datetime, timezone, timedelta
+from pathlib import Path
+from typing import List, Optional, Dict, Any
+import logging
+import time
+import json
```

Ruff (0.12.2)

`json` imported but unused

Remove unused import: `json`

(F401)